.pre-commit-config.yaml

```diff
@@ -0,0 +1,36 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-json
+      - id: check-ast
+      - id: check-shebang-scripts-are-executable
```

Comment:
Should we add the inverse, as well? `check-executables-have-shebangs`

`check-merge-conflict` might be another useful one to include.

And for sanity, maybe this one... `forbid-new-submodules` 🤣